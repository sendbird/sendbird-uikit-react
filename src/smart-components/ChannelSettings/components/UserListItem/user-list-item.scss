@import '../../../../styles/variables';

.sendbird-user-list-item--small {
  display: block;
  position: relative;
  box-sizing: border-box;
  height: 40px;

  &__avatar {
    cursor: pointer;
    position: absolute;
    top: 10px;
    left: 12px;
  }

  .sendbird-muted-avatar {
    position: absolute;
    top: 10px;
    left: 12px;
    z-index: 2;
    pointer-events: none;
  }

  &__title {
    position: absolute;
    top: 10px;
    left: 54px;
    width: 160px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }

  &__subtitle {
    position: absolute;
    top: 23px;
    left: 140px;
    max-width: 237px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }

  &__checkbox {
    position: absolute;
    top: 16px;
    right: 16px;
  }

  &__action {
    display: none;
    position: absolute;
    right: 10px;
    top: 4px;
  }

  &__operator {
    position: absolute;
    right: 24px;
    top: 10px;
  }

  &:hover,
  &.sendbird-icon--pressed {
    @include themed() {
      background-color: t(bg-1);
    }
    .sendbird-user-list-item--small__action {
      display: block;
    }

    .sendbird-user-list-item--small__operator {
      display: none;
    }
  }
  &.sendbird-icon--pressed {
    .sendbird-iconbutton {
      @include themed() {
        background-color: t(primary--1-bg--7);
      }
    }
  }
}
