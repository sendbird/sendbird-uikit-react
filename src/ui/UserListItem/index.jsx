import React, { useContext } from 'react';
import PropTypes from 'prop-types';

import { UserProfileContext } from '../../lib/UserProfileContext';
import { LocalizationContext } from '../../lib/LocalizationContext';
import Avatar from '../Avatar/index';
import MutedAvatarOverlay from '../Avatar/MutedAvatarOverlay';
import Checkbox from '../Checkbox';
import UserProfile from '../UserProfile';
import ContextMenu, { MenuItems } from '../ContextMenu';
import Label, { LabelTypography, LabelColors } from '../Label';

import './index.scss';

export default function UserListItem({
  className,
  user,
  checkBox,
  disableMessaging,
  currentUser,
  checked,
  onChange,
  action,
}) {
  const uniqueKey = user.userId;
  const actionRef = React.useRef(null);
  const parentRef = React.useRef(null);
  const avatarRef = React.useRef(null);
  const {
    disableUserProfile,
    renderUserProfile,
  } = useContext(UserProfileContext);
  const { stringSet } = useContext(LocalizationContext);
  return (
    <div
      className={[
        ...(Array.isArray(className) ? className : [className]),
        'sendbird-user-list-item',
      ].join(' ')}
      ref={parentRef}
    >
      {
        user.isMuted && (
          <MutedAvatarOverlay height={40} width={40} />
        )
      }
      <ContextMenu
        menuTrigger={(toggleDropdown) => (
          <Avatar
            className="sendbird-user-list-item__avatar"
            ref={avatarRef}
            src={user.profileUrl}
            width="40px"
            height="40px"
            onClick={() => {
              if (!disableUserProfile) {
                toggleDropdown();
              }
            }}
          />
        )}
        menuItems={(closeDropdown) => (
          <MenuItems
            openLeft
            parentRef={avatarRef}
            // for catching location(x, y) of MenuItems
            parentContainRef={avatarRef}
            // for toggling more options(menus & reactions)
            closeDropdown={closeDropdown}
            style={{ paddingTop: 0, paddingBottom: 0 }}
          >
            {
              renderUserProfile
                ? renderUserProfile({
                  user,
                  currentUserId: currentUser,
                  close: closeDropdown,
                })
                : (
                  <UserProfile
                    disableMessaging={disableMessaging}
                    user={user}
                    currentUserId={currentUser}
                    onSuccess={closeDropdown}
                  />
                )
            }
          </MenuItems>
        )}
      />
      <Label
        className="sendbird-user-list-item__title"
        type={LabelTypography.SUBTITLE_1}
        color={LabelColors.ONBACKGROUND_1}
      >
        {user.nickname || stringSet.NO_NAME}
        {
          (currentUser === user.userId) && (
            ' (You)'
          )
        }
      </Label>
      { // if there is now nickname, display userId
        !user.nickname && (
          <Label
            className="sendbird-user-list-item__subtitle"
            type={LabelTypography.CAPTION_3}
            color={LabelColors.ONBACKGROUND_2}
          >
            {user.userId}
          </Label>
        )
      }
      {
        checkBox && (
          // eslint-disable-next-line jsx-a11y/label-has-associated-control
          <label
            className="sendbird-user-list-item__checkbox"
            htmlFor={uniqueKey}
          >
            <Checkbox
              id={uniqueKey}
              checked={checked}
              onChange={(event) => onChange(event)}
            />
          </label>
        )
      }
      {
        user.role === 'operator' && (
          <Label
            className="sendbird-user-list-item__operator"
            type={LabelTypography.SUBTITLE_2}
            color={LabelColors.ONBACKGROUND_2}
          >
            Operator
          </Label>
        )
      }
      {
        action && (
          <div
            className="sendbird-user-list-item__action"
            ref={actionRef}
          >
            { action({ actionRef, parentRef })}
          </div>
        )
      }
    </div>
  );
}

UserListItem.propTypes = {
  className: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.arrayOf(PropTypes.string),
  ]),
  user: PropTypes.shape({
    userId: PropTypes.string,
    role: PropTypes.string,
    isMuted: PropTypes.bool,
    nickname: PropTypes.string,
    profileUrl: PropTypes.string,
  }).isRequired,
  disableMessaging: PropTypes.bool,
  currentUser: PropTypes.string,
  action: PropTypes.oneOfType([
    PropTypes.element,
    PropTypes.func,
  ]),
  checkBox: PropTypes.bool,
  checked: PropTypes.bool,
  onChange: PropTypes.func,
};

UserListItem.defaultProps = {
  className: '',
  currentUser: '',
  checkBox: false,
  disableMessaging: false,
  checked: false,
  action: null,
  onChange: () => { },
};
