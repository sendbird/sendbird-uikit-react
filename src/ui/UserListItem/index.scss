@import '../../styles/variables';

.sendbird-user-list-item {
  display: block;
  position: relative;
  box-sizing: border-box;
  width: 432px;
  height: 56px;
  @include themed() {
    border-bottom: 1px solid t(on-bg-4);
    background-color: t(bg--0-5);
  }

  .sendbird-user-list-item__avatar {
    position: absolute;
    top: 8px;
    left: 0px;
    z-index: 1;
    cursor: pointer;
  }

  .sendbird-muted-avatar {
    position: absolute;
    top: 8px;
    left: 0px;
    z-index: 2;
    pointer-events: none;
  }

  .sendbird-user-list-item__title {
    position: absolute;
    top: 17px;
    left: 56px;
    max-width: 250px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }

  .sendbird-user-list-item__subtitle {
    position: absolute;
    top: 23px;
    left: 140px;
    max-width: 237px;
    white-space: nowrap;
    overflow: hidden;
    text-overflow: ellipsis;
  }

  .sendbird-user-list-item__checkbox {
    position: absolute;
    top: 16px;
    right: 16px;
  }

  .sendbird-user-list-item__action {
    position: absolute;
    right: 0;
    top: 10px;
  }

  .sendbird-user-list-item__operator {
    position: absolute;
    right: 40px;
    top: 22px;
    &.checkbox {
      right: 66px;
    }
  }
}
