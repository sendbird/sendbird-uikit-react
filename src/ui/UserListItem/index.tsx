import React, { ChangeEvent, MutableRefObject, ReactElement, useContext } from 'react';
import { User } from '@sendbird/chat';
import { Member } from '@sendbird/chat/groupChannel';
import './index.scss';

import { UserProfileContext } from '../../lib/UserProfileContext';
import { LocalizationContext } from '../../lib/LocalizationContext';
import Avatar from '../Avatar/index';
import MutedAvatarOverlay from '../Avatar/MutedAvatarOverlay';
import Checkbox from '../Checkbox';
import UserProfile from '../UserProfile';
import ContextMenu, { MenuItems } from '../ContextMenu';
import Label, { LabelTypography, LabelColors } from '../Label';

export interface UserListItemProps {
  user: User;
  className?: string;
  checked?: boolean;
  checkBox?: boolean;
  isOperator?: boolean;
  disabled?: boolean;
  disableMessaging?: boolean;
  currentUser?: string;
  action?({ actionRef, parentRef }: {
    actionRef: MutableRefObject<any>,
    parentRef?: MutableRefObject<any>,
  }): ReactElement;
  onChange?(e: ChangeEvent<HTMLInputElement>): void;
  avatarSize?: string;
}

export default function UserListItem({
  user,
  className,
  checked,
  checkBox,
  isOperator,
  disabled,
  disableMessaging,
  currentUser,
  action,
  onChange,
  avatarSize = '40px',
}: UserListItemProps): ReactElement {
  const uniqueKey = user.userId;
  const actionRef = React.useRef(null);
  const parentRef = React.useRef(null);
  const avatarRef = React.useRef(null);
  const {
    disableUserProfile,
    renderUserProfile,
  } = useContext<UserProfileContext>(UserProfileContext);
  const { stringSet } = useContext(LocalizationContext);
  return (
    <div
      className={[
        ...(Array.isArray(className) ? className : [className]),
        'sendbird-user-list-item',
      ].join(' ')}
      ref={parentRef}
    >
      {
        (user as Member)?.isMuted && (
          <MutedAvatarOverlay height={40} width={40} />
        )
      }
      <ContextMenu
        menuTrigger={(toggleDropdown) => (
          <Avatar
            className="sendbird-user-list-item__avatar"
            ref={avatarRef}
            src={user.profileUrl}
            width={avatarSize}
            height={avatarSize}
            onClick={() => {
              if (!disableUserProfile) {
                toggleDropdown();
              }
            }}
          />
        )}
        menuItems={(closeDropdown) => (
          <MenuItems
            openLeft
            parentRef={avatarRef}
            // for catching location(x, y) of MenuItems
            parentContainRef={avatarRef}
            // for toggling more options(menus & reactions)
            closeDropdown={closeDropdown}
            style={{ paddingTop: '0px', paddingBottom: '0px' }}
          >
            {
              renderUserProfile
                ? renderUserProfile({
                  user,
                  currentUserId: currentUser,
                  close: closeDropdown,
                })
                : (
                  <UserProfile
                    disableMessaging={disableMessaging}
                    user={user}
                    currentUserId={currentUser}
                    onSuccess={closeDropdown}
                  />
                )
            }
          </MenuItems>
        )}
      />
      <Label
        className="sendbird-user-list-item__title"
        type={LabelTypography.SUBTITLE_1}
        color={LabelColors.ONBACKGROUND_1}
      >
        {user.nickname || stringSet.NO_NAME}
        {
          (currentUser === user.userId) && (
            ' (You)'
          )
        }
      </Label>
      { // if there is now nickname, display userId
        !user.nickname && (
          <Label
            className="sendbird-user-list-item__subtitle"
            type={LabelTypography.CAPTION_3}
            color={LabelColors.ONBACKGROUND_2}
          >
            {user.userId}
          </Label>
        )
      }
      {
        checkBox && (
          <label
            className="sendbird-user-list-item__checkbox"
            htmlFor={uniqueKey}
          >
            <Checkbox
              id={uniqueKey}
              checked={checked}
              disabled={disabled}
              onChange={(event) => onChange(event)}
            />
          </label>
        )
      }
      {
        isOperator && (
          <Label
            className={[
              'sendbird-user-list-item__operator',
              checkBox ? 'checkbox' : '',
            ].join(' ')}
            type={LabelTypography.SUBTITLE_2}
            color={LabelColors.ONBACKGROUND_2}
          >
            Operator
          </Label>
        )
      }
      {
        action && (
          <div
            className="sendbird-user-list-item__action"
            ref={actionRef}
          >
            {action({ actionRef, parentRef })}
          </div>
        )
      }
    </div>
  );
}
