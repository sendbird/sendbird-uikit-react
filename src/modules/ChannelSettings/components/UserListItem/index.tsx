import React, { ReactElement, useRef, useContext, ReactNode } from 'react';
import { User } from '@sendbird/chat';
import { type GroupChannel, Member } from '@sendbird/chat/groupChannel';

import './user-list-item.scss';

import { useSendbirdStateContext } from '../../../../lib/Sendbird';
import { LocalizationContext } from '../../../../lib/LocalizationContext';
import { UserProfileContext } from '../../../../lib/UserProfileContext';

import Avatar from '../../../../ui/Avatar/index';
import MutedAvatarOverlay from '../../../../ui/Avatar/MutedAvatarOverlay';
import Label, { LabelTypography, LabelColors } from '../../../../ui/Label';
import UserProfile from '../../../../ui/UserProfile';
import ContextMenu, { MenuItems } from '../../../../ui/ContextMenu';
import { UserListItemMenuProps } from '../../../../ui/UserListItemMenu/UserListItemMenu';

interface ActionProps {
  actionRef: React.RefObject<HTMLInputElement>;
  parentRef: React.RefObject<HTMLInputElement>;
}

interface Props {
  user: User | Member;
  channel: GroupChannel;
  /** @deprecated Doesn't need to fill this props */
  currentUser?: string;
  className?: string;
  /** @deprecated Use the props `renderListItemMenu` instead */
  action?(props: ActionProps): ReactNode;
  renderListItemMenu?: (props: UserListItemMenuProps) => ReactNode;
}

/**
 * This UserListItem component is usually used in the ChannelSettings.
 * There's another UserListItem which is used in the modal. (ui/UserListItem)
 */
const UserListItem = ({
  user,
  channel,
  className = '',
  action,
  renderListItemMenu,
}: Props): ReactElement => { const { config } = useSendbirdStateContext();
  const currentUser = config.userId;
  const actionRef = useRef(null);
  const parentRef = useRef(null);
  const avatarRef = useRef(null);
  const { stringSet } = useContext(LocalizationContext);
  const { disableUserProfile, renderUserProfile } = useContext(UserProfileContext);
  const injectingClassNames = Array.isArray(className) ? className : [className];
  return (
    <div
      ref={parentRef}
      className={[
        'sendbird-user-list-item--small', ...injectingClassNames,
      ].join(' ')}
    >
      {/* UserProfile */}
      <ContextMenu
        menuTrigger={(toggleDropdown) => (
          <>
            <Avatar
              onClick={() => {
                if (!disableUserProfile) {
                  toggleDropdown();
                }
              }}
              ref={avatarRef}
              className="sendbird-user-list-item--small__avatar"
              src={user.profileUrl}
              width={24}
              height={24}
            />
            {user instanceof Member && user.isMuted && (
              <MutedAvatarOverlay />
            )}
          </>
        )}
        menuItems={(closeDropdown) => (
          renderUserProfile
            ? renderUserProfile({
              user,
              currentUserId: currentUser ?? '',
              close: closeDropdown,
              avatarRef,
            })
            : (
              <MenuItems
                openLeft
                parentRef={avatarRef}
                // for catching location(x, y) of MenuItems
                parentContainRef={avatarRef}
                // for toggling more options(menus & reactions)
                closeDropdown={closeDropdown}
                style={{ paddingTop: '0px', paddingBottom: '0px' }}
              >
                <UserProfile
                  user={user}
                  currentUserId={currentUser}
                  onSuccess={closeDropdown}
                />
              </MenuItems>
            )
        )}
      />
      <Label
        className="sendbird-user-list-item--small__title"
        type={LabelTypography.SUBTITLE_1}
        color={LabelColors.ONBACKGROUND_1}
      >
        {user.nickname || stringSet.NO_NAME}
        {
          (currentUser === user.userId) && (
            stringSet.CHANNEL_SETTING__MEMBERS__YOU
          )
        }
      </Label>
      { // if there is now nickname, display userId
        !user.nickname && (
          <Label
            className="sendbird-user-list-item--small__subtitle"
            type={LabelTypography.CAPTION_3}
            color={LabelColors.ONBACKGROUND_2}
          >
            {user.userId}
          </Label>
        )
      }
      {
        user instanceof Member && user.role === 'operator' && (
          <Label
            className="sendbird-user-list-item--small__operator"
            type={LabelTypography.SUBTITLE_2}
            color={LabelColors.ONBACKGROUND_2}
          >
            {stringSet.CHANNEL_SETTING__MEMBERS__OPERATOR}
          </Label>
        )
      }
      {/* Deprecated logic */}
      {(!renderListItemMenu && action) && (
        <div ref={actionRef} className="sendbird-user-list-item--small__action">
          {action({ actionRef, parentRef })}
        </div>
      )}
      {renderListItemMenu && (
        <div ref={actionRef} className="sendbird-user-list-item--small__action">
          {renderListItemMenu({ channel, user })}
        </div>
      )}
    </div>
  );
};

export default UserListItem;
