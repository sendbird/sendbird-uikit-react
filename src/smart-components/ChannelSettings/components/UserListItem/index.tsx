import React, { ReactElement, useRef, useContext } from 'react';

import SendBird from 'sendbird';

import { LocalizationContext } from '../../../../lib/LocalizationContext';
import { UserProfileContext } from '../../../../lib/UserProfileContext';
import Avatar from '../../../../ui/Avatar/index';
import MutedAvatarOverlay from '../../../../ui/Avatar/MutedAvatarOverlay';
import Label, { LabelTypography, LabelColors } from '../../../../ui/Label';
import UserProfile from '../../../../ui/UserProfile';
import ContextMenu, { MenuItems } from '../../../../ui/ContextMenu';

import './user-list-item.scss';

const COMPONENT_NAME = 'sendbird-user-list-item--small';

interface ActionProps {
  actionRef: React.RefObject<HTMLInputElement>;
  parentRef: React.RefObject<HTMLInputElement>;
}

type CustomUser = SendBird.User & {
  isMuted: boolean;
  role: string;
};

interface Props {
  user: CustomUser;
  currentUser?: string;
  className?: string;
  action?(props: ActionProps): ReactElement;
}

const UserListItem = ({
  user,
  className,
  currentUser,
  action,
}: Props): ReactElement => {
  const actionRef = useRef(null);
  const parentRef = useRef(null);
  const avatarRef = useRef(null);
  const { stringSet } = useContext(LocalizationContext);
  const {
    disableUserProfile,
    renderUserProfile,
  } = useContext(UserProfileContext);
  const injectingClassNames = Array.isArray(className) ? className : [className];
  return (
    <div
      ref={parentRef}
      className={[
        COMPONENT_NAME, ...injectingClassNames,
      ].join(' ')}
    >
      {
        user.isMuted && (
          <MutedAvatarOverlay />
        )
      }
      <ContextMenu
        menuTrigger={(toggleDropdown) => (
          <Avatar
            onClick={() => {
              if (!disableUserProfile) {
                toggleDropdown();
              }
            }}
            ref={avatarRef}
            className={`${COMPONENT_NAME}__avatar`}
            src={user.profileUrl}
            width={24}
            height={24}
          />
        )}
        menuItems={(closeDropdown) => (
          <MenuItems
            openLeft
            parentRef={avatarRef}
            // for catching location(x, y) of MenuItems
            parentContainRef={avatarRef}
            // for toggling more options(menus & reactions)
            closeDropdown={closeDropdown}
            style={{ paddingTop: 0, paddingBottom: 0 }}
          >
            {
              renderUserProfile
                ? renderUserProfile({
                  user,
                  currentUserId: currentUser,
                  close: closeDropdown,
                })
                : (
                  <UserProfile
                    user={user}
                    currentUserId={currentUser}
                    onSuccess={closeDropdown}
                  />
                )
            }
          </MenuItems>
        )}
      />
      <Label
        className={`${COMPONENT_NAME}__title`}
        type={LabelTypography.SUBTITLE_1}
        color={LabelColors.ONBACKGROUND_1}
      >
        {user.nickname || stringSet.NO_NAME}
        {
          (currentUser === user.userId) && (
            " (You)"
          )
        }
      </Label>
      { // if there is now nickname, display userId
        !user.nickname && (
          <Label
            className={`${COMPONENT_NAME}__subtitle`}
            type={LabelTypography.CAPTION_3}
            color={LabelColors.ONBACKGROUND_2}
          >
            {user.userId}
          </Label>
        )
      }
      {
        user.role === 'operator' && (
          <Label
            className={`${COMPONENT_NAME}__operator`}
            type={LabelTypography.SUBTITLE_2}
            color={LabelColors.ONBACKGROUND_2}
          >
            Operator
          </Label>
        )
      }
      {
        action && (
          <div ref={actionRef} className={`${COMPONENT_NAME}__action`}>
            { action({ actionRef, parentRef }) }
          </div>
        )
      }
    </div>
  );
}

export default UserListItem;
