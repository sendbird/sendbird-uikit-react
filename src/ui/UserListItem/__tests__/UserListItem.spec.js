import React from 'react';
import { render } from '@testing-library/react';

import UserListItem from "../index";

const getUserList = () => [
  {
    userId: 'userid-1',
    nickname: 'Topy',
    profileUrl: 'https://static.sendbird.com/sample/profiles/profile_12_512px.png',
  },
  {
    userId: 'userid-2',
    nickname: 'Middly',
    profileUrl: 'https://static.sendbird.com/sample/user_sdk/user_sdk_01.png',
  },
  {
    userId: 'userid-3',
    nickname: 'Bottomy',
    profileUrl: 'https://static.sendbird.com/sample/user_sdk/user_sdk_14.png',
  },
];

describe('ui/UserListItem', () => {
  it.skip('should render text prop', function () {
    const [user1] = getUserList();
    render(<UserListItem user={user1} />);

    // expect(
    //   component.find(".sendbird-user-list-item__title").hostNodes().text()
    // ).toEqual(user1.nickname);
  });

  it('should do a snapshot test of the UserListItem DOM', function () {
    const [user1] = getUserList();
    const { asFragment } = render(
      <UserListItem
        user={user1}
        checkBox
        checked={true}
        onChange={() => { }}
      />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
