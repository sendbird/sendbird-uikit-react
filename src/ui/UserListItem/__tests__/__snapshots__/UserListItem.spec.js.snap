// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ui/UserListItem should do a snapshot test of the UserListItem DOM 1`] = `
<DocumentFragment>
  <div
    class=" sendbird-user-list-item"
  >
    <div
      class="sendbird-context-menu"
      style="display: inline;"
    >
      <div
        class="sendbird-user-list-item__avatar sendbird-avatar"
        role="button"
        style="height: 40px; width: 40px; z-index: 0; left: 0px;"
        tabindex="0"
      >
        <div
          class="sendbird-avatar-img sendbird-image-renderer"
          style="width: 100%; min-width: min(400px, 40px); max-width: 400px;"
        >
          <div
            class="sendbird-image-renderer__image"
            style="width: 100%; min-width: min(400px, 40px); max-width: 400px; position: absolute; background-repeat: no-repeat; background-position: center; background-size: cover; background-image: url(https://static.sendbird.com/sample/profiles/profile_12_512px.png);"
          />
          <img
            alt=""
            class="sendbird-image-renderer__hidden-image-loader"
            src="https://static.sendbird.com/sample/profiles/profile_12_512px.png"
          />
        </div>
      </div>
    </div>
    <span
      class="sendbird-user-list-item__title sendbird-label sendbird-label--subtitle-1 sendbird-label--color-onbackground-1"
    >
      Topy
    </span>
    <label
      class="sendbird-user-list-item__checkbox"
      for="userid-1"
    >
      <label
        class="sendbird-checkbox "
        for="userid-1"
      >
        <input
          checked=""
          id="userid-1"
          type="checkbox"
        />
        <span
          class="sendbird-checkbox--checkmark "
        />
      </label>
    </label>
  </div>
</DocumentFragment>
`;
